config/v1/types_infrastructure.go

```diff
@@ -19,39 +18,54 @@ type Infrastructure struct {
 	Status InfrastructureStatus `json:"status"`
 }
 
+// InfrastructureSpec contains settings that apply to the cluster infrastructure.
 type InfrastructureSpec struct {
 	// secret reference?
 	// configmap reference to file?
 }
 
+// InfrastructureStatus describes the infrastructure the cluster is leveraging.
 type InfrastructureStatus struct {
 	// platform is the underlying infrastructure provider for the cluster. This
 	// value controls whether infrastructure automation such as service load
 	// balancers, dynamic volume provisioning, machine creation and deletion, and
 	// other integrations are enabled. If None, no infrastructure automation is
-	// enabled.
+	// enabled. Allowed values are "AWS", "Azure", "GCP", "Libvirt",
+	// "OpenStack", "VSphere", and "None". Individual components may not support
```

Comment:
I think we need advice for components to indicate their lack of support.

Reply:
We do, but that'll be in #1112.  I don't want our public api docs to be instructions to our teams (beyond general statements), especially since we don't know the instructions quite yet.